src/nl/hannahsten/texifyidea/settings/conventions/TexifyConventionsConfigurable.kt

```diff
@@ -0,0 +1,216 @@
+            }
+        }
+
+        val nameColumnInfo =
+            object : TableModelEditor.EditableColumnInfo<LabelConvention, LabelConvention>("Element") {
+                override fun valueOf(item: LabelConvention): LabelConvention = item
+                override fun isCellEditable(item: LabelConvention?): Boolean = false
+                override fun getColumnClass(): Class<*> = LabelConvention::class.java
+                override fun getRenderer(item: LabelConvention?): TableCellRenderer {
+                    return object : DefaultTableCellRenderer() {
+                        override fun getTableCellRendererComponent(
+                            table: JTable?,
+                            value: Any?,
+                            selected: Boolean,
+                            focus: Boolean,
+                            row: Int,
+                            column: Int
+                        ): Component {
+                            val convention = value as LabelConvention?
+                            super.getTableCellRendererComponent(table, convention?.name, selected, focus, row, column)
+                            //noinspection unchecked
+                            icon = if (value != null) {
+                                when (value.type!!) {
+                                    LabelConventionType.ENVIRONMENT -> TexifyIcons.DOT_ENVIRONMENT
+                                    LabelConventionType.COMMAND -> TexifyIcons.DOT_COMMAND
+                                }
+                            }
+                            else {
+                                null
+                            }
+
+                            return this
+                        }
+                    }
+                }
+            }
+
+        val enabledColumnInfo =
+            object : TableModelEditor.EditableColumnInfo<LabelConvention, Boolean>("Should have label") {
+                override fun getColumnClass(): Class<*> = Boolean::class.java
+                override fun getTooltipText(): String =
+                    "If enabled, TeXiFy issues a warning if the given Latex element does not have a label"
+
+                override fun valueOf(item: LabelConvention): Boolean = item.enabled
+                override fun setValue(item: LabelConvention, value: Boolean) {
+                    item.enabled = value
+                }
+
+                override fun getMaxStringValue(): String? {
+                    // Give a hint that the column won't ever need more space than its label
+                    return super.getName()
+                }
+            }
+
+        val itemEditor = object : CollectionItemEditor<LabelConvention> {
+            override fun getItemClass(): Class<out LabelConvention> = LabelConvention::class.java
+
+            override fun clone(item: LabelConvention, forInPlaceEditing: Boolean): LabelConvention {
+                TODO("Not yet implemented")
```

Comment:
to do

Reply:
I changed it to a simple exception. It is not implemented on purpose though (see above)